drivers/at: add function to read raw data bytes from modem

fedepell opened the pull request:
### Contribution description
This PR adds a function to read raw data bytes from the modem device. All previous functions work on a line basis and are not suitable for some operations like raw data bytes transfer.
My use case for example is transfering via GPRS an image file to flash remotely the device. To do this I start the firmware image transfer by first issuing AT commands and then having to read the raw data bytes and flashing on the device (following concepts from #9342 ).

### Testing procedure
The manual test under tests/drivers_at was augmented to call also this function. With a device with a modem connected use on the test shell for example:
`send_recv_bytes AT+CGMM 31`
The function will send the AT+CGMM and then try to read 31 bytes. If less are found it will wait for the timeout to expire, otherwise return immediately with the 31 bytes.

fedepell commented:
I'm using more this function now as dealing with binary data transfers over modem (ie. AT#SRECV commands to transfer via sockets). Could it be possible to review and see if this could be suitable for merging into master?  @kaspar030 maybe?

Thanks a lot!

kaspar030 commented on drivers/at/at.c:
```diff
@@ -58,6 +58,26 @@ void at_send_bytes(at_dev_t *dev, const char *bytes, size_t len)
+{
+    char *resp_pos = bytes;
+
+    memset(bytes, 0, len);
```
My gut feeling says, drop this.

fedepell replied:
Ok :)

fedepell commented on drivers/include/at.h:
```diff
@@ -220,6 +220,18 @@ int at_expect_bytes(at_dev_t *dev, const char *bytes, uint32_t timeout);
+ * @param[in]   len     maximum number of bytes to receive
+ * @param[in]   timeout timeout (in usec) of inactivity to finish read
+ *
+ * @returns     Number of bytes read, eventually 0
```
Done, added management with -ETIMEDOUT if no data after timeout.

fedepell commented on tests/driver_at/main.c:
```diff
@@ -105,6 +105,25 @@ static int send_lines(int argc, char **argv)
+
+    ssize_t len = at_recv_bytes(&at_dev, buffer, atoi(argv[2]), 10 * US_PER_SEC);
+
+    printf("Response (len=%d): %s\n", (int)len, buffer);
```
Added and tested!

fedepell commented:
@kaspar030 : many thanks for reviewing! Hope it's in better shape now, I can in case squash when looks fine!

fedepell commented:
@kaspar030 : thanks for the inputs! Ok I reverted the ETIMEDOUT part and left out the memset and squashed all together. I just tried to make the "eventually" comment a bit clear :)

Thanks again for the review!

fedepell commented:
@kaspar030 : I put as "resolved" the changes some time ago but I believe (sorry if I'm wrong :) ) that this should be somehow confirmed by you, correct? Thanks!

kaspar030 commented:
Could someone give this a quick spin?

fedepell commented:
@aabadie : could we have this in 2019.01 if we find someone that can try it a little out? thanks!

aabadie commented:
It's added to 2019.01. Since there's already an (Untested) ACK, it is very close from being merged. I won't have the time and am not sure how to test this.

But maybe @leandrolanzieri could help ?

leandrolanzieri commented:
Nice. Yes! I can try it out with the R410M modem

fedepell commented:
Thanks @leandrolanzieri for testing!

For the record: I use it with a Telit GL865-DUAL V3 from a saml21 based hardware.
